microsoft-edge/devtools-guide-chromium/memory-problems/heap-snapshots.md

```diff
@@ -263,47 +235,54 @@ To find an object in the collected heap, you can search using **Ctrl+F** and giv
 <!-- ====================================================================== -->
 ## Uncover DOM leaks
 
-The heap profiler has the ability to reflect bidirectional dependencies between browser native objects (DOM nodes, CSS rules) and JavaScript objects.  This helps to discover otherwise invisible leaks happening due to forgotten detached DOM subtrees floating around.
+The heap profiler has the ability to reflect bidirectional dependencies between browser native objects (DOM nodes, CSS rules) and JavaScript objects.  This helps to discover otherwise invisible leaks that happen because of forgotten detached DOM subtrees that remain in memory.
 
-DOM leaks may be bigger than you think.  Consider the following sample.  When is the `#tree` garbage-collected?
+Consider the following code sample.  When is the element with ID `tree` garbage-collected?
 
 ```javascript
+// Getting a reference to the #tree element.
 var treeRef = document.querySelector("#tree");
+
+// Getting a reference to the #leaf element,
+// which is a descendant of the #tree element.
 var leafRef = document.querySelector("#leaf");
-var body = document.querySelector("body");
 
-body.removeChild(treeRef);
+// Removing the #tree element from the DOM.
+document.body.removeChild(treeRef);
+
+// The #tree element can't be garbage-collected yet
+// because the treeRef variable still exists.
 
-//#tree in not GC yet due to treeRef
+// Removing the treeRef variable.
 treeRef = null;
 
-//#tree is not GC yet due to indirect reference from leafRef
+// The #tree element is still not garbage-collected
+// because of the indirect reference from the leafRef variable.
 
+// Removing the leafRef variable.
 leafRef = null;
-//#NOW can be #tree GC
+
+// Now the #tree element can be garbage-collected.
 ```
 
```

Comment:
above, we introducted the code block by posing a question (a dubious way of writing, but works):

When is the element with ID `tree` garbage-collected?

Here, we must explicitly repeat that phrasing and now deliver the answer in a way that makes it clear the question and the answer to that question.
```suggestion
When is the element with ID `tree` garbage-collected?  The above code indicates that the element with ID `tree` garbage-collected when X.
```
manually apply this change

Reply:
Re-worked the entire section to avoid asking the question.